components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/DefaultRoleManagementListener.java

```diff
@@ -0,0 +1,510 @@
+            if (APPLICATION_AUDIENCE.equalsIgnoreCase(roleBasicInfo.getAudience())) {
+                roleBasicInfo.setAudienceName(getApplicationName(roleBasicInfo.getAudienceId(), tenantDomain));
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean preGetRoleIdListOfUser(String userId, String tenantDomain) throws IdentityRoleManagementException {
+
+        return true;
+    }
+
+    @Override
+    public boolean postGetRoleIdListOfUser(List<String> roleIds, String userId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return true;
+    }
+
+    @Override
+    public boolean preGetRoleIdListOfGroups(List<String> groupIds, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return true;
+    }
+
+    @Override
+    public boolean postGetRoleIdListOfGroups(List<String> roleIds, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return true;
+    }
+
+    @Override
+    public boolean preGetRoleIdListOfIdpGroups(List<String> groupIds, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return true;
+    }
+
+    @Override
+    public boolean postGetRoleIdListOfIdpGroups(List<String> roleIds, List<String> groupIds,
+                                                String tenantDomain) throws IdentityRoleManagementException {
+
+        return true;
+    }
+
+    @Override
+    public boolean preDeleteRolesByApplication(String applicationId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return true;
+    }
+
+    @Override
+    public boolean postDeleteRolesByApplication(String applicationId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return true;
+    }
+
+    /**
+     * Validate application role audience.
+     *
+     * @param applicationId Application Id.
+     * @throws IdentityRoleManagementException Error occurred while validating application role audience.
+     */
+    private void validateApplicationRoleAudience(String applicationId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        try {
+            ServiceProvider app = ApplicationManagementService.getInstance()
+                    .getApplicationByResourceId(applicationId, tenantDomain);
+            if (app == null) {
+                throw new IdentityRoleManagementClientException(INVALID_AUDIENCE.getCode(),
+                        "Invalid audience. No application found with application id: " + applicationId);
+            }
+            boolean valid = false;
+            for (ServiceProviderProperty property : app.getSpProperties()) {
+                // TODO :  use osgi service to get this
+                if (ALLOWED_ROLE_AUDIENCE_PROPERTY_NAME.equals(property.getName())) {
+                    if (APPLICATION_AUDIENCE.equalsIgnoreCase(property.getValue())) {
```

Comment:
can combine conditions in two if clauses using &&

Reply:
fixed